src/Distributed/multi_architectures.jl

```diff
@@ -181,27 +178,32 @@ function decrement_index(i, R, topo)
     end
 end
 
-function RankConnectivity(model_index, ranks, topology)
-    i, j, k = model_index
+function RankConnectivity(local_index, ranks, topology)
+    i, j, k = local_index
     Rx, Ry, Rz = ranks
     TX, TY, TZ = topology
 
     i_east  = increment_index(i, Rx, TX)
     i_west  = decrement_index(i, Rx, TX)
     j_north = increment_index(j, Ry, TY)
     j_south = decrement_index(j, Ry, TY)
-    k_top   = increment_index(k, Rz, TZ)
-    k_bot   = decrement_index(k, Rz, TZ)
 
-    r_east  = isnothing(i_east)  ? nothing : index2rank(i_east, j, k, Rx, Ry, Rz)
-    r_west  = isnothing(i_west)  ? nothing : index2rank(i_west, j, k, Rx, Ry, Rz)
+    r_east  = isnothing(i_east)  ? nothing : index2rank(i_east,  j, k, Rx, Ry, Rz)
+    r_west  = isnothing(i_west)  ? nothing : index2rank(i_west,  j, k, Rx, Ry, Rz)
     r_north = isnothing(j_north) ? nothing : index2rank(i, j_north, k, Rx, Ry, Rz)
     r_south = isnothing(j_south) ? nothing : index2rank(i, j_south, k, Rx, Ry, Rz)
-    r_top   = isnothing(k_top)   ? nothing : index2rank(i, j, k_top, Rx, Ry, Rz)
-    r_bot   = isnothing(k_bot)   ? nothing : index2rank(i, j, k_bot, Rx, Ry, Rz)
 
-    return RankConnectivity(east=r_east, west=r_west, north=r_north,
-                            south=r_south, top=r_top, bottom=r_bot)
+    r_northeast = isnothing(i_east) || isnothing(j_north) ? nothing : index2rank(i_east, j_north, k, Rx, Ry, Rz)
```

Comment:
should `r_northeast` be something like `northeastern_rank`? Not sure what `r` is.